packages/gasket-plugin-docs/lib/index.d.ts

```diff
@@ -8,21 +9,80 @@ declare module '@gasket/engine' {
     }
   }
 
-  // TODO: correctly document what the full payload is...
-  type DocsConfigSet = {
-    docsRoot: string
-  };
+  export interface DocsSetupModulesConfig {
+    [key: string]: DocsSetup
+  }
+
+  export interface DocsTransformHandlerData {
+    filename: String,
```

Comment:
You should use the TS "native" types `string`, `boolean`, and `number`, not the boxed versions of them; otherwise you won't be able to assign literal values.

```suggestion
    filename: string,
```